### PR TITLE
add_check_CKV_AWS_103

### PR DESCRIPTION
**Pull request by livnoni**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/SubnetPublicIP.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class SubnetPublicIP(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure VPC subnets do not assign a public IP by default"
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/SubnetPublicIP.py:7]: @livnoni Should be `assign public IPs by default`

File: checkov/terraform/checks/resource/aws/SubnetPublicIP.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        key="map_public_ip_on_launch"
+        if key in conf.keys():
+            if not conf[key][0]:
+                return CheckResult.PASSED
+            return CheckResult.FAILED
+        else:
+            return CheckResult.PASSED
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/SubnetPublicIP.py:20]: @livnoni Looks like a `BaseResourceValueCheck` to me

File: checkov/terraform/checks/resource/aws/SubnetPublicIP.py (hunk 1 above)
Comment (livnoni) [reply checkov/terraform/checks/resource/aws/SubnetPublicIP.py:7]: got it

File: checkov/terraform/checks/resource/aws/SubnetPublicIP.py (hunk 2 above)
Comment (livnoni) [reply checkov/terraform/checks/resource/aws/SubnetPublicIP.py:20]: r u sure?
it's a bit more complicated than true\false

Comment (livnoni) [reply checkov/terraform/checks/resource/aws/SubnetPublicIP.py:20]: Thanks nimrod, using BaseResourceNegativeValueCheck instaed of BaseResourceCheck

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**